devtools/cli/src/main/java/io/quarkus/cli/build/GradleRunner.java

```diff
@@ -58,13 +61,32 @@ public BuildTool getBuildTool() {
     }
 
     @Override
-    public Integer listExtensions(RunModeOption runMode, ListFormatOptions format, boolean installable, String searchPattern) {
+    public Integer listExtensionCategories(RunModeOption runMode, CategoryListFormatOptions format) throws Exception {
+        ArrayDeque<String> args = new ArrayDeque<>();
+        setGradleProperties(args, runMode.isBatchMode());
+
+        args.add("listCategories");
+        args.add("--fromCli");
+        if (StringUtils.isNotBlank(format.getFormatString())) {
+            args.add("--format=" + format.getFormatString());
+        }
+        return run(prependExecutable(args));
+    }
+
+    @Override
+    public Integer listExtensions(RunModeOption runMode, ListFormatOptions format, boolean installable, String searchPattern,
+            String category) {
         ArrayDeque<String> args = new ArrayDeque<>();
         setGradleProperties(args, runMode.isBatchMode());
 
         args.add("listExtensions");
         args.add("--fromCli");
-        args.add("--format=" + format.getFormatString());
+        if (StringUtils.isNotBlank(format.getFormatString())) {
+            args.add("--format=" + format.getFormatString());
+        }
+        if (StringUtils.isNotBlank(category)) {
```

Comment:
This is another: Either String.isBlank or a test against null (as with search pattern)